Doc/whatsnew/3.14.rst

```diff
@@ -999,6 +999,25 @@ urllib
   supporting SHA-256 digest authentication as specified in :rfc:`7616`.
   (Contributed by Calvin Bui in :gh:`128193`.)
 
+* Improve support for ``file:`` URLs.
+
+  In :func:`urllib.request.url2pathname`:
```

Comment:
I think we should be more explicit that `url2pathname` now performs network requests via `socket`. Both to say we better confom to the RFC, but also as readers might reasonably assume the function is entirely lexical in nature. Notably, [`socket.gethostbyname{_ex}`](https://docs.python.org/3/library/socket.html#socket.gethostbyname) raises an audit event.

I don't mind if this is recorded in *What's New* as well as the module documentation, but I think it should be recorded as an observable behaviour change.

Reply:
I've mentioned that we call `socket.gethostbyname()` in the `.. versionchanged::` directive - do you think that's explicit enough? Happy to go even more explicit!